### PR TITLE
Issue2/gpl 633 event warehouse should have same recovery and error handling of unified wh

### PR DESCRIPTION
**Pull request by JamesGlover**
Closes #

Changes proposed in this pull request:

*
*
* ...

### CODE REVIEW

**Comment by JamesGlover** [comment]
There's a demo Sequencescape branch here https://github.com/JamesGlover/sequencescape/tree/warren_experimentation_branch but it currently runs warren from 'path' so the tests will be failing on GH. Its mostly to show how warren integrates into Sequencescape.

**Comment by JamesGlover** [comment]
Not done:
- Unfied warehouse has a 'delayed_requeue' functionality, which lets messages get plonked on an alternative queue with a ttl, that then dead-letters them back onto the main queue. This shouldn't be too hard to add in to this architecture, as it pretty much just follows the pattern of a dead-letter queue registration. The delay requeue method should be easily adaptable from unified warehouse.

https://github.com/sanger/unified_warehouse/blob/5c61d2bd3099617212c848d231af0effd942701b/lib/postman/message.rb#L89-L100

**Comment by JamesGlover** [comment]
There's also a bit of adaptation that needs to be done to the deployment project as the means of invoking the consumers has changed.

`bundle exec warren consumer start`

Also, we're no longer daemonised, and don't have pid files. Fortunately the monit docs provide a nice way to work with processes under systemd without pid files now.

File: lib/warren/app/config.rb [hunk 1]
```diff
@@ -7,13 +7,18 @@ class Config
       # We keep the template as plain text as it allows us to add comments
       TEMPLATE = <<~TEMPLATE
```
Comment (JamesGlover) [inline lib/warren/app/config.rb:8]: Ideally I'd move this out into a template file like I did with the subscriber class files.

File: README.md [hunk 2]
```diff
@@ -14,26 +14,105 @@ gem 'sanger-warren'
...
-    $ bundle install
+```bash
+bundle install
+```
 
 Or install it yourself as:
 
-    $ gem install sanger-warren
+```bash
+gem install sanger-warren
```
Comment (stevieing) [inline README.md:24]: I think this should be sanger_warrren

**Review by stevieing — APPROVED** [review]

**Pull request merged**